core/src/main/java/com/linecorp/armeria/internal/annotation/ResponseConversionUtil.java

```diff
@@ -0,0 +1,275 @@
+    /**
+     * Returns a mutable {@link HttpHeaders} which contains the same headers as the specified {@code headers}.
+     */
+    public static HttpHeaders toMutableHeaders(HttpHeaders headers) {
+        requireNonNull(headers, "headers");
+        return headers.isImmutable() ? HttpHeaders.copyOf(headers) : headers;
+    }
+
+    /**
+     * Returns a new {@link HttpResponseWriter} which has a content converted from the collected objects.
+     *
+     * @param stream a sequence of objects
+     * @param headers to be written to the returned {@link HttpResponseWriter}
+     * @param trailingHeaders to be written to the returned {@link HttpResponseWriter}
+     * @param contentConverter converts the collected objects into a content of the response
+     * @param executor executes the collecting job
+     */
+    public static HttpResponseWriter aggregateFrom(Stream<?> stream,
+                                                   HttpHeaders headers, HttpHeaders trailingHeaders,
+                                                   Function<Object, HttpData> contentConverter,
+                                                   Executor executor) {
+        requireNonNull(stream, "stream");
+        requireNonNull(headers, "headers");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+        requireNonNull(contentConverter, "contentConverter");
+        requireNonNull(executor, "executor");
+
+        return aggregateFrom(collectFrom(stream, executor),
+                             headers, trailingHeaders, contentConverter);
+    }
+
+    /**
+     * Returns a new {@link HttpResponseWriter} which has a content converted from the collected objects.
+     *
+     * @param publisher publishes objects
+     * @param headers to be written to the returned {@link HttpResponseWriter}
+     * @param trailingHeaders to be written to the returned {@link HttpResponseWriter}
+     * @param contentConverter converts the collected objects into a content of the response
+     */
+    public static HttpResponseWriter aggregateFrom(Publisher<?> publisher,
+                                                   HttpHeaders headers, HttpHeaders trailingHeaders,
+                                                   Function<Object, HttpData> contentConverter) {
+        requireNonNull(publisher, "publisher");
+        requireNonNull(headers, "headers");
+        requireNonNull(trailingHeaders, "trailingHeaders");
+        requireNonNull(contentConverter, "contentConverter");
+
+        return aggregateFrom(collectFrom(publisher),
+                             headers, trailingHeaders, contentConverter);
+    }
+
+    private static HttpResponseWriter aggregateFrom(CompletableFuture<?> future,
+                                                    HttpHeaders headers, HttpHeaders trailingHeaders,
+                                                    Function<Object, HttpData> contentConverter) {
+        final HttpResponseWriter writer = HttpResponse.streaming();
```

Comment:
Curious if `HttpResponse.from()` is a better option here. By changing so, all `aggregateFrom()` methods could return `HttpResponse` as well.

Reply:
I wanted to decorate a response with `ExceptionFilteredHttpResponse` on [here](https://github.com/line/armeria/pull/1530/files#diff-767b436cc8c0427e4457c63752a88bd2R303) only if the returned type is instance of `HttpResponseWriter`.
Is it better to use `HttpResponse.from()`?